packages/world-vercel/src/queue.ts

```diff
@@ -26,13 +28,22 @@ export function createQueue(): Queue {
   const createQueueHandler: Queue['createQueueHandler'] = (prefix, handler) => {
     return handleCallback({
       [`${prefix}*`]: {
-        default: (body, meta) => {
+        default: async (body, meta) => {
           const { payload, queueName } = MessageWrapper.parse(body);
-          return handler(payload, {
+          const result = await handler(payload, {
             queueName,
             messageId: MessageId.parse(meta.messageId),
             attempt: meta.deliveryCount,
           });
+          if (typeof result?.timeoutSeconds === 'number') {
+            // For Vercel Queue, enforce the max visibility limit:
+            //   - When a step function throws a `RetryableError`, the retryAfter timestamp is updated and stored on the Step document
+            result.timeoutSeconds = Math.min(
+              result.timeoutSeconds,
+              VERCEL_QUEUE_MAX_VISIBILITY
+            );
+          }
```

Comment:
Should we add a debug log here?

Reply:
A bit tricky at the moment because this logic is in `world-vercel` but the logger is in `core`.